### PR TITLE
Add assertions to TPU accelerator for device availability

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

Add assertions to TPU accelerator for device availability
This aims to simplify the accelerator connector logic and rewrite effort in https://github.com/PyTorchLightning/pytorch-lightning/pull/11448

This would move the assertion logic from Trainer constructor checks to runtime at trainer.fit/validate/test/predict calls

Related PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/11797

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by four4fish — APPROVED** [review]

File: pytorch_lightning/accelerators/tpu.py [hunk 1]
```diff
@@ -25,6 +26,15 @@
 class TPUAccelerator(Accelerator):
     """Accelerator for TPU devices."""
 
+    def setup_environment(self, root_device: torch.device) -> None:
```
Comment (carmocca) [inline pytorch_lightning/accelerators/tpu.py:29]: Move to __init__?

**Review by carmocca — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -98,6 +98,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added a `_Stateful` support for `LightningDataModule` ([#11637](https://github.com/PyTorchLightning/pytorch-lightning/pull/11637))
 
 
+- Added checks to `TPUAccelerator.setup_environment` to assert device availability ([#11799](https://github.com/PyTorchLightning/pytorch-lightning/pull/11799))
```
Comment (rohitgr7) [inline CHANGELOG.md:101]:
```suggestion
- Added checks to `TPUAccelerator` to assert device availability ([#11799](https://github.com/PyTorchLightning/pytorch-lightning/pull/11799))
```

File: pytorch_lightning/accelerators/tpu.py [hunk 3]
```diff
@@ -25,6 +26,15 @@
 class TPUAccelerator(Accelerator):
     """Accelerator for TPU devices."""
 
+    def __init__(self) -> None:
+        """
+        Raises:
+            MisconfigurationException:
+                If the TPU device is not available.
```
Comment (rohitgr7) [inline pytorch_lightning/accelerators/tpu.py:33]:
```suggestion
                If the `torch_xla` is not installed.
```

File: pytorch_lightning/accelerators/tpu.py [hunk 4]
```diff
@@ -25,6 +26,15 @@
...
+        """
+        if not _XLA_AVAILABLE:
```
Comment (rohitgr7) [inline pytorch_lightning/accelerators/tpu.py:35]: should we add another one for TPU_AVAILABLE?
https://github.com/PyTorchLightning/pytorch-lightning/issues/3104

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/accelerators/tpu.py [hunk 5]
```diff
@@ -25,6 +26,15 @@
...
+            raise MisconfigurationException("The TPU Accelerator requires torch_xla and a TPU device to run.")
```
Comment (tchaton) [inline pytorch_lightning/accelerators/tpu.py:36]:
```suggestion
            raise MisconfigurationException("The TPU Accelerator requires the `torch_xla` library and `TPU devices` to run.")
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by ananthsub** [comment]
Closing out in favor of the approach in #11797

**Pull request closed without merging**